contrib/unbound.service.in

```diff
@@ -73,7 +73,7 @@ SystemCallFilter=~@clock @cpu-emulation @debug @keyring @module mount @obsolete
 RestrictNamespaces=yes
 LockPersonality=yes
 RestrictSUIDSGID=yes
-ReadWritePaths=@UNBOUND_RUN_DIR@ @UNBOUND_CHROOT_DIR@
+ReadWritePaths=@UNBOUND_CHROOT_DIR@ @UNBOUND_ROOTKEY_FILE@
```

Comment:
`UNBOUND_RUN_DIR` should stay for reasons stated above.

As you pointed out in https://github.com/NLnetLabs/unbound/issues/475 `UNBOUND_ROOTKEY_FILE` won't add write access to whole dir where key is stored when it's changed from default `UNBOUND_RUN_DIR/root.key`. When it's not changed then it isn't needed as `UNBOUND_RUN_DIR` is already covered so adding `UNBOUND_CHROOT_DIR` here doesn't fix anything really.

Reply:
I'm not an expert in this field, but it appears to me that chrooting into a directory that is writable to the user you are trying to constrain is a potential security issue. Here's a StackExchange post that explains the issue better than I could:
https://unix.stackexchange.com/a/332571

This tl;dr is: If an attacker manages to exploit a vulnerability in the unbound process and run malicious code within the context of that process, (s)he could potentially gain persistence across daemon restarts by creating specially a crafted `/etc` and/or `/lib` and/or `/usr` within the chroot directory.

Reply:
Right, chroot is considered as weak security boundary but using it is default unbound behavior. The service doesn't make anything worse here.

Reply:
We definitely need to use ` TemporaryFileSystem=/:ro` without `ProtectSystem` - I think the system would be again mounted over a TemporaryFileSystem as a read-only. I'll test this configuration and will try to come up with the solution.